### PR TITLE
Fix incorrectly creating a union with one member as the inferred parameter type with tuples

### PR DESCRIPTION
**Pull request by MaryamZi**
## Purpose
$title, this was causing invalid analysis elsewhere where we assume a union to have > 1 member.

Fixes https://github.com/ballerina-platform/ballerina-lang/issues/39680

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Review by hasithaa — APPROVED** [review]
Looks good to me.

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeParamAnalyzer.java [hunk 1]
```diff
@@ -580,8 +580,9 @@ private void findTypeParamInTupleForArray(Location loc, BArrayType expType, BTup
         if (actualType.restType != null) {
             tupleTypes.add(actualType.restType);
         }
-        BUnionType tupleElementType = BUnionType.create(null, tupleTypes);
-        findTypeParam(loc, expType.eType, tupleElementType, env, resolvedTypes, result);
+
+        BType type = tupleTypes.size() == 1 ? tupleTypes.iterator().next() : BUnionType.create(null, tupleTypes);
```
Comment (chiranSachintha) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeParamAnalyzer.java:584]: It is not valid to create a union type when the size of the `tupleTypes` array is zero. Attempting to do so will cause the following program to crash. I have created a separate [issue](https://github.com/ballerina-platform/ballerina-lang/issues/39699) for this problem
```ballerina
import ballerina/io;

const mm = "s";
public function main() {
    var data = [].map( m => m + 1);
    io:println(data);
}
```

Comment (chiranSachintha) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeParamAnalyzer.java:584]: When the program uses the 'string:char' type, it compiles without any errors. 
```ballerina
import ballerina/io;

public function main() {
    string:Char o = "s";
    var data = ["10", o].map( m => m + 1);
    io:println(data); // prints ["101","s1"]
}
```
I have created a separate [issue](https://github.com/ballerina-platform/ballerina-lang/issues/39700) for this problem

Comment (MaryamZi) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeParamAnalyzer.java:584]: This isn't an anonymous function issue - see https://github.com/ballerina-platform/ballerina-lang/issues/39700#issuecomment-1441108180.

**Pull request merged**